### PR TITLE
fix: create team form

### LINKED ISSUE #7771: [CAL-1274] New team creation issue
Found a bug? Please fill out the sections below. 👍

### Issue Summary

When attempting to create a new team, if the team link stub is unavailable, a newly entered
stub is not reassessed thus preventing the continuation of the team creation.

### Steps to Reproduce
https://www.loom.com/share/93ca578936b3473cb180f9d80a7d9a72

<sub>[CAL-1274](https://linear.app/calcom/issue/CAL-1274/new-team-creation-issue)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/7771
2) delete button in button group

https://user-images.githubusercontent.com/53316345/227203519-8ba0ceb8-dea0-484d-b391-0c97c3f7f38f.mov

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: packages/features/ee/teams/components/CreateANewTeamForm.tsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ export const CreateANewTeamForm = () => {
                     if (newTeamFormMethods.formState.touchedFields["slug"] === undefined) {
                       newTeamFormMethods.setValue("slug", slugify(e?.target.value));
                     }
+                    createTeamMutation.reset();
```
Comment (PeerRich) [inline packages/features/ee/teams/components/CreateANewTeamForm.tsx:78]: what is this for @Udit-takkar ?

**Review by PeerRich — APPROVED** [review]

File: packages/ui/components/button/Button.tsx [hunk 2]
```diff
@@ -135,7 +135,7 @@ const buttonClasses = cva(
         disabled: [false, undefined],
         color: "destructive",
         className:
-          "border dark:text-white text-gray-900 hover:text-red-700 focus-visible:text-red-700 dark:hover:text-red-700 dark:focus-visible:text-red-700 hover:border-red-100 focus-visible:border-red-100 hover:bg-red-100  focus-visible:bg-red-100 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-red-700",
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/ui/components/button/Button.tsx:138]: oh ok, I think it is just for button group. I was checking here: https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS?node-id=323-21417&t=2b8ZLFnrsCZeKha0-0

**Review by emrysal — APPROVED** [review]
Merging, asked feedback from @Jaibles though so may need a follow up 🙏  (I believe he would like to feedback his thoughts on the placement etc. of the error message)

**Comment by Udit-takkar** [comment]
@Jaibles okay will change the error message UI here.

**Pull request merged**